src/overlays/actors/ovl_Bg_Po_Event/z_bg_po_event.c

```diff
@@ -21,47 +41,576 @@ const ActorInit Bg_Po_Event_InitVars = {
+        this->timer = 0;
+        Audio_PlayActorSound2(thisx, NA_SE_EN_PO_LAUGH);
+        this->actionFunc = BgPoEvent_PaintingVanish;
+    } else if (this->collider.base.acFlags & 2) {
+        if (!BgPoEvent_NextPicture(this)) {
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_PO_SISTERS, thisx->posRot.pos.x,
+                        thisx->posRot.pos.y - 40.0f, thisx->posRot.pos.z, 0, thisx->shape.rot.y, 0,
+                        thisx->params + ((this->actorType - 1) << 8));
+            func_800800F8(globalCtx, 0xC58, 0x50, thisx, 0);
+            func_80078884(0x4802);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A7328.s")
+        } else {
+            Audio_PlayActorSound2(thisx, NA_SE_EN_PO_LAUGH2);
+            func_800800F8(globalCtx, 0xC58, 0x23, thisx, 0);
+        }
+        if (thisx->parent != NULL) {
+            thisx->parent->child = NULL;
+            thisx->parent = NULL;
+        }
+        if (thisx->child != NULL) {
+            thisx->child->parent = NULL;
+            thisx->child = NULL;
+        }
+        this->timer = 20;
+        this->actionFunc = BgPoEvent_PaintingBurn;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A7444.s")
+void BgPoEvent_PaintingBurn(BgPoEvent* this, GlobalContext* globalCtx) {
+    Vec3f sp54;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A7500.s")
+    this->timer--;
+    sp54.x = (Math_Sins(this->dyna.actor.shape.rot.y) * 5.0f) + this->dyna.actor.posRot.pos.x;
+    sp54.y = Math_Rand_CenteredFloat(66.0f) + this->dyna.actor.posRot.pos.y;
+    sp54.z = Math_Rand_CenteredFloat(50.0f) + this->dyna.actor.posRot.pos.z;
+    if (this->timer >= 0) {
+        if (this->actorType == 2) {
+            func_8002A6B8(globalCtx, &sp54, &zeroVec, &zeroVec, 0x64, 0, 0xFF, 0xFF, 0x96, 0xAA, 0xFF, 0x00, 0x00, 1, 9,
+                          1);
+        } else {
+            func_8002A6B8(globalCtx, &sp54, &zeroVec, &zeroVec, 0x64, 0, 0xC8, 0xFF, 0xFF, 0xAA, 0x32, 0x64, 0xFF, 1, 9,
+                          1);
+        }
+    }
+    if (this->timer == 0) {
+        this->dyna.actor.draw = NULL;
+    }
+    if (this->timer < -60) {
+        Actor_Kill(&this->dyna.actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A7530.s")
+void BgPoEvent_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoEvent* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A7568.s")
+    this->actionFunc(this, globalCtx);
+    if ((this->actionFunc == BgPoEvent_AmyWait) || (this->actionFunc == BgPoEvent_PaintingPresent)) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Po_Event/func_808A75B8.s")
+Gfx* displayLists[] = { D_060075A0, D_060079E0, D_06006830, D_06006D60, D_06007230 };
```

Comment:
since you put that here can it also be moved into Draw?

Reply:
No. In fact, since there's no inlines it doesn't even need to be moved down, so I could put it at the top with the rest of the data if that's preferred.